dvc/cache.py

```diff
@@ -86,19 +116,51 @@ def __getitem__(self, key):
         return self._items[key]
 
     def add(self, scheme, checksum, name):
-        self._items[scheme][checksum].add(name)
+        """Add a mapped name for the specified checksum."""
+        self._items[scheme][checksum].names.add(name)
+
+    def add_child_cache(self, checksum, cache, suffix=""):
+        """Add/update child cache for the specified checksum."""
+        for scheme, src in cache._items.items():
+            dst = self._items[scheme][checksum].children
```

Comment:
Do we need to check if `checksum` exists?

Reply:
Each nested level is a defaultdict, so if `checksum` isn't already in our cache, it will be populated with the correct empty containers when we reference it here

Reply:
Oops, sorry. Forgot about that. :(